### PR TITLE
Add relu kernel for metal

### PR DESCRIPTION
**Pull request by jbochi**
Also improves the error message when a kernel is missing.

Diff is somewhat big because I sorted lines. Let me know if you prefer to leave as it is.

With this change, we can run t5 in metal:

```
$ RUST_BACKTACE=1 cargo run --example t5 --release --features metal -- --model-id "t5-small" --prompt "translate English to German: As much as six inches of rain could fall in the New York City region through Monday morning, and officials warned of flooding along the coast." --temperature 0 --decode
    Finished release [optimized] target(s) in 0.19s
     Running `target/release/examples/t5 --model-id t5-small --prompt 'translate English to German: As much as six inches of rain could fall in the New York City region through Monday morning, and officials warned of flooding along the coast.' --temperature 0 --decode`
 Bis zu sechs Zoll Regen könnte in der Region New York bis Montag morgens fallen, und Beamte warnten vor Überschwemmungen entlang der Küste.
34 tokens generated (132.73 token/s)
```

### CODE REVIEW

File: candle-core/src/metal_backend.rs [hunk 1]
```diff
@@ -678,33 +680,33 @@ impl BackendStorage for MetalStorage {
...
-                ("ulog", DType::F16) => strided::log::HALF,
-                ("ugelu", DType::F16) => strided::gelu::HALF,
-                ("ugelu_erf", DType::F16) => strided::gelu_erf::HALF,
-                ("uerf", DType::F16) => strided::erf::HALF,
-                ("uceil", DType::F16) => strided::ceil::HALF,
-                ("ufloor", DType::F16) => strided::floor::HALF,
-                ("uround", DType::F16) => strided::round::HALF,
-                (name, dtype) => crate::bail!("Match {name} - {dtype:?}"),
+                ("usqrt", DType::F32) => strided::sqrt::FLOAT,
+                (name, dtype) => crate::bail!("No binary kernel in metal for {name} - {dtype:?}"),
```
Comment (grzuy) [inline candle-core/src/metal_backend.rs:709]: Two cents here, this is "strided unary", not "binary" :-)

Comment (grzuy) [reply candle-core/src/metal_backend.rs:709]: Tried to improve more generally all these in https://github.com/huggingface/candle/pull/1491.

Comment (jbochi) [reply candle-core/src/metal_backend.rs]: Ooops. Thanks! I copied the messages you proposed in your PR to avoid merge conflicts.

**Review by Narsil — COMMENTED** [review]
Can you remove the ordering of ops into another PR, it's adding noise which might hide a subtle bug.

Just add relu in this PR. (Totally ok to create another PR to reorder if you really want it)

**Comment by jbochi** [comment]
> ve the ordering of ops into another PR, it's adding noise which might hide a subtle bug.
> 
> Just add relu in this PR. (Totally ok to create another PR to reorder

Done. PTAL.

And please squash the commits when merging. 😅

**Comment by Narsil** [comment]
No worries, squash is the default :).

Can you do the `cargo fmt` ? I'll merge afterwards.

**Review by Narsil — APPROVED** [review]
LGTM (After cargo fmt).

**Comment by jbochi** [comment]
Thanks! I ran rustfmt and resolved the new merge conflict.

Happy new year!

**Comment by jbochi** [comment]
Can you please take a look again?

Thanks!

**Review by Narsil — APPROVED** [review]

**Pull request merged**